src/main/java/uk/nhs/tis/trainee/notifications/service/NotificationService.java

```diff
@@ -86,29 +127,24 @@ public void execute(JobExecutionContext jobExecutionContext) {
   }
 
   /**
-   * Save a Programme Updated sent-notification into the history.
+   * Get the user account details from Cognito (if they have signed-up to TIS Self-Service), or from
+   * the Trainee Details profile if not.
    *
-   * @param jobDetails The JobDataMap.
-   * @param email      The recipient email address.
-   * @param sentAt     The instant the email was sent.
-   * @return The saved notification history.
+   * @param personId The person ID to search for.
+   * @return The user account details, or null if not found.
    */
-  private History saveNotificationSent(JobDataMap jobDetails, String email, Instant sentAt) {
-    ObjectId objectId = ObjectId.get(); //TODO let DB generate this?
-    RecipientInfo recipientInfo
-        = new RecipientInfo(jobDetails.getString(TIS_ID), MessageType.EMAIL, email);
-    NotificationType notificationType = NotificationType.valueOf(
-        jobDetails.get("notificationType").toString());
-
-    TemplateInfo templateInfo
-        = new TemplateInfo(notificationType.getTemplateName(), "v1.0.0",
-        jobDetails.getWrappedMap());
-    TisReferenceInfo tisReference = new TisReferenceInfo(TisReferenceType.PROGRAMME_MEMBERSHIP,
-        jobDetails.get(TIS_ID).toString());
-
-    History history = new History(objectId, tisReference, notificationType, recipientInfo,
-        templateInfo, sentAt);
-
-    return historyService.save(history);
+  private UserAccountDetails getAccountDetails(String personId) {
+    try {
+      return emailService.getRecipientAccount(personId);
+    } catch (IllegalArgumentException e) {
+      //no TSS account (or duplicate accounts), so try to get from trainee-details profile.
```

Comment:
Not sure if duplicate accounts should be handled differently here. We'd need a way to distinguish these from the `getReceiptAccount()` function.